wordpress-mobile/WordPress-Android
# Jetpack App Basics: Handle sites

Issue #14299

This is a draft PR for Jetpack app sites handling. Based on the Jetpack or WordPress app build variant, it controls:
1. "Connect another site" visibility on  the login epilogue
2. Jetpack sites filtering on the My Sites tab -> Site Picker
3. "Add new site" menu visibility on the My Sites tab -> Site Picker

`SiteStore.fetchSites` is accessed from different classes mentioned below: 
- **WordPress class**

    - `WordPress.java`
    - `JetpackConnectionResultActivity.java`
    - `SitePickerActivity.java`

- **Login Lib classes**

    - `LoginBaseFormFragment`
    - `LoginWpcomService`

- **FluxC Example Test classes**

    - `MockedStack_SiteTest`
    - `ReleaseStack_ActivityLogTestJetpack`
    - `ReleaseStack_InsightsTestJetpack`
    - `ReleaseStack_ManageInsightsTestJetpack`
    - `ReleaseStack_MediaTestJetpack`
    - `ReleaseStack_PluginTestJetpack`
    - `ReleaseStack_PostSchedulingTestJetpack`
    - `ReleaseStack_ScanTestJetpack`
    - `ReleaseStack_SiteTestJetpack`
    - `ReleaseStack_SiteTestWPCom`
    - `ReleaseStack_ThemeTestJetpack`
    - `ReleaseStack_TimeStatsTestJetpack`
    - `ReleaseStack_WPComBase`

- **FluxC Example app**
    - `SiteSelectorDialog`
    - `SiteActionBuilder`
    - `MainInstafluxActivity`

Instead of introducing a new boolean param in the `SiteStore.fetchSites` for jetpack filtering and setting it from the WordPress or  Login modules based on the build variant, FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1936 adds the `filters=jetpack` [request param](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/fac74b5d124b0c8670706b28e77339ba0d7ceab6/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.java#L158) in the `SiteRestClient` if it has a Jetpack app context. Note that [app context already exists](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.java#L145-L147) in the class constructor.

Going this route, we don't need to make any changes to the login lib for this task and also lesser number of classes will be affected since we're not adding new param to the `fetchSites` method.

@zwarm, I discussed with you already, here's how it will look. 
@zwarm, @ParaskP7 let me know what you think of it. I'll proceed accordingly.  

**Pre test**

You must update your local copy of `google-services.json` before running the app.
-- Copy existing `org.wordress.android`, `org.wordress.android.prealpha`, `org.wordress.android.beta` sections and replace with `com.jetpack.android`,` com.jetpack.android.prealpha`, `com.jetpack.android.beta`.

**To Test**

Build and launch below product flavors:
`wordpressJalapenoDebug`
`jetpackJalapenoDebug`

Verify that for Jetpack build variant:
- "Connect another site" is removed from the login epilogue
- Only "jetpack" sites for the user  are shown on the My Sites tab -> Site Picker
- Create a new site ability is disabled on the My Sites tab -> Site Picker

Verify that for WordPress build variant:
- "Connect another site" is present on the login epilogue
- All sites for the user are shown on the My Sites tab -> Site Picker
- Create a new site ability is present on the My Sites tab -> Site Picker

**Merge Instructions**

This PR targets `feature/jetpack-app-basics`.
We'll decide next week if we want to split these changes for WordPress and Jetpack app based on how we'd like to merge corresponding FluxC PR. 
Till that time, do not merge this PR.

PR submission checklist:
